app-backend/batch/src/main/scala/com/azavea/rf/batch/airflow/export/CreateExportDef.scala

```diff
@@ -0,0 +1,148 @@
+package com.azavea.rf.batch.airflow.export
+
+import java.util.UUID
+
+import org.xbill.DNS._
+
+import scala.collection.JavaConverters._
+import scala.concurrent.{Await, Future}
+import scala.concurrent.duration._
+import scala.util._
+
+import akka.actor.ActorSystem
+import akka.stream.ActorMaterializer
+import cats.data._
+import cats.implicits._
+import com.amazonaws.auth.DefaultAWSCredentialsProviderChain
+import com.amazonaws.services.elasticmapreduce.{AmazonElasticMapReduceClient, AmazonElasticMapReduceClientBuilder}
+import com.amazonaws.services.elasticmapreduce.model.{AddJobFlowStepsRequest, StepConfig, HadoopJarStepConfig}
+import io.circe.generic.JsonCodec
+import io.circe.syntax._
+
+import com.azavea.rf.batch.Job
+import com.azavea.rf.batch.util._
+import com.azavea.rf.database.{Database => DB}
+import com.azavea.rf.database.tables._
+import com.azavea.rf.datamodel._
+
+case class CreateExportDef(exportId: UUID)(implicit val database: DB) extends Job {
+  val name = "create_export_def"
+    
+  protected def writeExportDefToS3(exportDef: ExportDefinition) = {
+    logger.info(s"Uploading export definition ${exportDef.id.toString} to S3 at ${exportDefConfig.bucketName}")
+    val uri = s"rasterfoundry-development-data-us-east-1/${exportDefConfig.bucketName}"
+    val future = Future[Option[String]] {
+      S3.putObject(uri, s"${exportDef.id.toString}.json", None, exportDef.asJson.toString)
+      Some(uri)
+    }
+
+    future onComplete {
+      case Success(_) => {
+        logger.info(s"Export definition uploaded to S3 at ${uri}")
+      }
+      case Failure(ex) => {
+        logger.error(s"An error occurred uploading export defintion to S3 for ${exportDef.id}")
+        logger.error(ex.stackTraceString)
+      }
+    }
+
+    future
+  }
+
+  protected def getClusterId() = {
+    val dnsLookup = new Lookup(exportDefConfig.awsDataproc, Type.TXT, DClass.IN)
+    dnsLookup.run()
+    val clusterId = dnsLookup.getAnswers().map(_.rdataToString).head.replace("\"", "")
+    clusterId
+  }
+
+  protected def startExportEmrJob(exportDef: ExportDefinition, exportDefUri: String): Unit = {
+    val jarStep = new HadoopJarStepConfig()
+    jarStep.setJar(jarPath)
+    jarStep.setArgs(List("/usr/bin/spark-submit", "--master", "yarn", "--deploy-mode",
+          "cluster", "--class", "com.azavea.rf.batch.export.Export", "--driver-memory", "2G",
+          "s3://rasterfoundry-global-artifacts-us-east-1/batch/rf-batch-b933688.jar",
```

Comment:
one minor thing: probably it makes sense to move all of these params into `application.conf`? in order not to hardcode values like memory, jar name, main class.